java/icicle/MemoryState.java

```diff
@@ -0,0 +1,186 @@
+import java.util.*;
+import java.time.LocalDate;
+
+class MemoryState<T> implements IcicleState<T>
```

Comment:
oh man. I wrote a php backend for the Disciple compiler. I'm still waiting for the right opportunity to really use it though
https://github.com/DDCSF/ddc/blob/master/packages/ddc-core-babel/DDC/Core/Babel/PHP.hs